cloud_blobstore/gs.py

```diff
@@ -58,7 +60,7 @@ def get_api_response(self, next_token=None):
         return resp
 
     def get_listing_from_response(self, resp):
-        return (b.name for b in resp)
+        return ((b.name, b) for b in resp) if self.verbose else (b.name for b in resp)
```

Comment:
I think this is a pretty confusing API. That the return type is `typing.Iterable[typing.Union[typing.Tuple[str, typing.Any], str]]` is not a great sign.

Reply:
I see what you mean. This was my attempt at keeping it DRY, however, I see that it would be better to introduce a new function and extracted shared logic behind the scenes.